src/daemon/Display.cpp

```diff
@@ -149,10 +149,11 @@ namespace SDDM {
         connect(this, &Display::loginFailed, m_socketServer, &SocketServer::loginFailed);
         connect(this, &Display::loginSucceeded, m_socketServer, &SocketServer::loginSucceeded);
 
-        connect(m_greeter, &Greeter::failed,
-                QCoreApplication::instance(), [] {
-                    QCoreApplication::instance()->exit(23);
-                });
+        connect(m_greeter, &Greeter::failed, this, &Display::stop);
+        connect(m_greeter, &Greeter::ttyFailed, this, [this] {
+            VirtualTerminal::jumpToVt(1, true);
```

Comment:
Why tty1? In openSUSE we have a getty running there, this might break it

Reply:
getty is everywhere in my experience. Do you have an alternative in mind?

Reply:
Would you prefer having a default tty setting?

Reply:
And on FreeBSD we have getty running on ttys `0-7` and X server starts on `8`.

Reply:
On openSUSE it's getty on tty1-tty6 by default (started on the first switch) and X on tty7 (sddm.conf has `MinimumVT=7`).

Because getty is started on demand, tty1-tty6 are actually free until the switch and something like OPENQRY can return them for use by e.g. `kwin_wayland`.

It's not entirely clear to me what the VT number specified here effectively does...

Reply:
> On openSUSE it's getty on tty1-tty6 by default (started on the first switch) and X on tty7 (sddm.conf has MinimumVT=7).

MinimumVT was dropped a while ago. https://github.com/sddm/sddm/commit/f0b6dec66a6fc8cc6e21da57f81190843b483f76

> Because getty is started on demand, tty1-tty6 are actually free until the switch and something like OPENQRY can return them for use by e.g. kwin_wayland.

Well that kind of is the problem. We sometimes want to have a predictable tty for our session and the greeter. openqry can tell us which one is free but getty takes them over and never releases them so you can easily end up with a session that is all over the place.

> It's not entirely clear to me what the VT number specified here effectively does...

Where you want the graphical sessions to start from. I guess not unlike the MinimumVT setting from back then.

Reply:
> In openSUSE we have a getty running there, this might break it

Unless you were patching out this `Conflicts=getty@tty1.service` in the service file, this statement is not true. If you were patching it, I imagine it was conflicting at times?

In any case, I've added a CMake variable so that you can use start from a different tty.

Reply:
> Unless you were patching out this Conflicts=getty@tty1.service in the service file, this statement is not true. If you were patching it, I imagine it was conflicting at times?

Yep (https://build.opensuse.org/package/view_file/openSUSE:Factory/sddm/0001-Systemd-service-unit-Use-tty7-by-default.patch?expand=1)! Though we don't actually use `sddm.service` by default, we have `xdm.service` which starts the displaymanager (configured in `/etc/sysconfig/displaymanager) itself. That'll probably change soon though.